include/ghex/transport_layer/ucx/context.hpp

```diff
@@ -166,25 +164,22 @@ namespace gridtools {
                     // make shared worker
                     // use single-threaded UCX mode, as per developer advice
                     // https://github.com/openucx/ucx/issues/4609
-                    m_worker = worker_type(this, &m_thread_primitives, nullptr, UCS_THREAD_MODE_SINGLE);
-                    
+                    m_worker = worker_type{this, m_mutex, UCS_THREAD_MODE_SINGLE};
+
                     // intialize database
                     m_db.init(m_worker.address());
                 }
-                
+
                 communicator_type get_serial_communicator()
                 {
                     return {&m_worker,&m_worker};
                 }
 
-                communicator_type get_communicator(const thread_token& t)
+                communicator_type get_communicator()
```

Comment:
same here for the above serial communicator - I think it is not needed anymore

Reply:
I agree